static/docs/user-guide/contributing.md

```diff
@@ -112,6 +112,127 @@ To pass additional arguments:
     $ python -m tests --pdb
 ```
 
+## Testing remotes
+
+This tests require additional effort to set up, so they are skipped by default.
+You don't need this in most cases, however, if you develop or fix some remote
+related code you might need to go through steps below.
+
+Install requirements for whatever remote(s) you are going to test:
+
+```bash
```

Comment:
- let's use `dvc` instead of `bash` for code blocks that show commands to run.
- let's remove artificial indents - engine is fixed to indent it automatically.
- let's add `$ ` before each command. engine is fixed to avoid the on user select. It helps to highlight commands properly.